src/sentry/static/sentry/app/stores/globalSelectionStore.jsx

```diff
@@ -29,18 +30,44 @@ const getDefaultSelection = () => {
   };
 };
 
-const isValidSelection = (selection, organization) => {
-  const allowedProjects = new Set(
-    organization.projects.filter(project => project.isMember).map(p => parseInt(p.id, 10))
-  );
-  if (
-    Array.isArray(selection.projects) &&
-    selection.projects.some(project => !allowedProjects.has(project))
-  ) {
-    return false;
-  }
+const getProjectsByIds = async (organization, projectIds) => {
+  const api = new Client();
```

Comment:
You could pass the api instance in from the component that calls this function.

Reply:
hmmm this function is called by `isValidSelection` which is called by `loadInitialData` which is called by `OrganizationContext`. I made the change, but wondering if there is an easier way to give access to the api instance rather than passing it through multiple function calls

Reply:
I think it's better to have the store a bit more decoupled from the component. I don't think we would necessarily want api requests associated with a store to be reset when the component unmounts. Thoughts @markstory ?

Reply:
Good point about the unmount case. I didn't think of that.